### ISSUE TITLE
Problem when trying to generate new fingerprints

### ISSUE THREAD

**Issue by jat9292** [spoilt333/onco-aae#4]
Hello,
When trying to generate new fingerprints, I sample 640 normal vectors in hidden layer like you did, after training, but then I don't get any generated log concentration < -5... least I get is around -2.5 .. 
Maybe there is a problem with the training, because I get the following losses after 50 epochs:
disc: 0.384376, enc_fp : 1.444782, mani_fp: 0.000000, enc_tgi: 3018.004395, dec_fp : 0.691891, dec_conc : 2.271482
As you can notice, encoder tgi loss is way higher than the other losses, and it stays around 3000 during the whole training.  Why is this happening? Is it normal?
Also, I noticed that in your code, you put:
self.enc_loss = self.enc_fp_loss
but , shouldn't we use instead the following expression:  self.enc_loss = self.enc_fp_loss+self.enc_tgi_loss ?
Also, unrelated question, but could you tell me why you add an absolute value inside the exponential in the discriminator and ecnoder losses?
Thank you in advance.

**Comment by spoilt333** [comment]
"disc: 0.384376" looks like AAE didn't converge. In general both D and G losses should be -ln(2), however it isn't necessary.
I can't examine your losses because last time I trained this network was last year:) may it be that enc_tgi calculated as sum instead of mean? dec_pf looks like random. So I suppose that something went wrong.
"self.enc_loss = self.enc_fp_loss" to answer this I should review code, but may be enc_tgi_loss used somewhere else?

"Also, unrelated question, but could you tell me why you add an absolute value inside the exponential in the discriminator and ecnoder losses?" this losses are just reformulated logloss, you can find examples and derivation in the internet.

**Comment by jat9292** [comment]
Hello,
Thank you for your answer!
However I would like to ask one more question concerning the generation of the new fingerprints:
- In the hidden encoded space , why did you choose to sample also the tgi neuron according to a centered normal distribution (according to your article) ? If we want an anticancer proprety, shouldn't we sample tgi between 0 (complete inhibition) and -1 (complete cell kill) ?

**Comment by spoilt333** [comment]
Due to TGI formula it it is not clear what actually means negative value. Moreover, it isn't a big deal to achieve -1 value if you use very toxic drug that can kill any cell. So, as a proof of concept we chose small positive values which means "almost stop growth"

**Comment by gaojunhui68** [comment]
@jat9292  
Hi , to generate new fingerprints , sample 640 normal vectors in hidden layer ...
Is this code included in the onco-aae? Or do you have to write it yourself?

Thanks.

**Comment by gaojunhui68** [comment]
@jat9292 would you describe how can we prepare the unique_data_file of 6252 .How many columns are there in this file? What's the data for each column?

I appreciate your help.
